articles/getting-started-with-nodemcu/index.md

```diff
@@ -0,0 +1,140 @@
+description: A tutorial on introduction to microcontroller, NodeMCU, and its uses in IoT.
+author: srishilesh-p-s
+date: 2021-01-00T00:00:00-00:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/getting-started-with-nodemcu/hero.jpg
+    alt: Getting started with NodeMCU
+---
+In this article, we will learn about the NodeMCU microcontroller. By the end of the article, you will have an overview of what a microcontroller is, what NodeMCU is, and its use in the Internet of Things. You will also learn step-by-step procedures for building a simple program to control the blinking of LEDs using NodeMCU.
+<!--more-->
+As a prerequisite, a little knowledge about [C programming language](https://en.wikipedia.org/wiki/C_(programming_language)) would help to follow this article along.
+
+### Table of contents
+- [Microcontroller](#microcontroller)
+- [NodeMCU](#nodemcu)
+- [Step by Step Guide](#step-by-step-guide)
+- [Conclusion](#conclusion)
+- [Further Reading](#further-reading)
+
+### Microcontroller
+> According to [Wikipedia](https://en.wikipedia.org/wiki/Microcontroller), A microcontroller (MCU for microcontroller unit) is a small computer on a single metal-oxide-semiconductor (MOS) integrated circuit (IC) chip. A microcontroller contains one or more CPUs (processor cores) along with memory and programmable input/output peripherals.
+
+In simple words, a Microcontroller can be called a mini-computer with an in-built CPU processor, Memory, and other I/O peripherals.
+
+Microcontrollers come with different configurations and architectures depending on the application it is used for. It is being used extensively in various industries like Manufacturing, Robotics, Automation, the Internet of Things, and so on.
+
+For example, Microcontrollers are used as embedded devices in appliances like ACs and Washing machines which enables humans to interact with the devices. They contain in-built programs for simplifying human tasks.
+
+### NodeMCU
+In this article, we will be focusing on the NodeMCU microcontroller. NodeMCU stands for Node MicroController Unit. The below image shows a NodeMCU microcontroller board.
+
+![NodeMCU ESP2866](/engineering-education/getting-started-with-nodemcu/nodemcu.png)
+
+[Image Source](https://cityos-air.readme.io/docs/esp8266-nodemcu)
+
+> According to [this](https://components101.com/development-boards/nodemcu-esp8266-pinout-features-and-datasheet) article, NodeMCU is an open-source Lua based firmware and development board specially targeted for IoT based applications. It includes firmware that runs on the ESP8266 Wi-Fi SoC from Espressif Systems, and hardware that is based on the ESP-12 module.
+
+#### Features
+- NodeMCU ESP12 module contains Tensilica Xtensa® 32-bit LX106 RISC microprocessor.
+- Operates at 80 to 160 MHz.
+- Operating voltage of 3.3V.
+- 128KB RAM and 4MB of Flash memory.
+- Contains inbuilt 802.11b/g/n WiFi transceiver.
+- Contains 17 GPIOs (General Purpose Input Output).
```

Comment:
Could you elaborate on what GPIO and UART is and explain? It would definitely add value to the article.